### DISCUSSION TITLE
OS dependent dependencies

### DISCUSSION THREAD

**Discussion by patrick-egenlauf** [astral-sh/rye#1392] [Q&A] [answered]
I use `jax[cuda12]` on my computer (Ubuntu). Since `jax[cuda12]` is not supported on Windows or MacOS, it is not possible to test my package on these systems with a GitHub workflow as described in #1375. Is there a way to somehow check the operating system first and install either `jax` or `jax[cuda12]` depending on the OS?

I found this discussion #981, however, I am not sure if this is what I am looking for. It would be nice to somehow include this in the `pyproject.toml` file.

**Comment by trappitsch** [comment, marked as answer]
You can specify platform-dependent dependencies directly in the `pyproject.toml`. Some more info [here](https://packaging.python.org/en/latest/specifications/dependency-specifiers/).

As an example, my `dependencies` in the `pyproject.toml` of one project looks something like this:

```toml
dependencies = [
    "click>=8.1.7",
    "dmgbuild>=1.6.1; sys_platform=='darwin'",
]
```

Here, `click` is installed on any platform, `dmgbuild` only if `sys_platform` is equal to `darwin`.

Hope this helps!

**Reply by patrick-egenlauf** [reply]
Thanks a lot!

Is there a way to check for a GPU as well? This is not relevant to the GitHub workflow, but if a user wants to install it on their local machine, `jax` should be installed if no GPU is available, even if the OS is `Linux`.

I only found it for `poetry` (see [this post](https://stackoverflow.com/questions/78467412/how-to-toggle-torch-cpu-and-gpu-installation-using-poetry)).
